hscontrol/types/node.go

```diff
@@ -190,6 +192,33 @@ func (node *Node) IPsAsString() []string {
 	return ret
 }
 
+func (node *Node) HostInfoAsProtoStruct() *structpb.Struct {
+	if node.HostinfoDatabaseField == "" {
+		return nil
+	}
+
+	var mapper map[string]interface{}
+	if err := json.Unmarshal([]byte(node.HostinfoDatabaseField), &mapper); err != nil {
```

Comment:
You should not rely on this field as it is only set when we save, and not when things change. `.Hostinfo` must be used, can you pass that straight to `structpb.NewStruct()`?